File: samples/cpp/src/rde_cpp/src/rde_subscriber.cpp
#include "rde_cpp/rde_subscriber.hpp"

RdeSubscriber::RdeSubscriber()
: Node("rde_subscriber")
{
  subscription_ = this->create_subscription<std_msgs::msg::String>(
    "rde", 10, std::bind(&RdeSubscriber::topic_callback, this, std::placeholders::_1));
}

void RdeSubscriber::topic_callback(const std_msgs::msg::String::SharedPtr msg) const
{
  RCLCPP_INFO(this->get_logger(), "I heard: '%s'", msg->data.c_str());
}

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<RdeSubscriber>());
  rclcpp::shutdown();
  return 0;
}

File: samples/py/src/rde_py/rde_py/rde_subscriber.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class RdeSubscriber(Node):
    def __init__(self):
        super().__init__('rde_subscriber')
        self.subscription = self.create_subscription(
            String,
            'rde',
            self.listener_callback,
            10)
        self.get_logger().info('RDE Subscriber node started')

    def listener_callback(self, msg):
        self.get_logger().info(f'Received: "{msg.data}"')


def main(args=None):
    rclpy.init(args=args)

    rde_subscriber = RdeSubscriber()

    try:
        rclpy.spin(rde_subscriber)
    except KeyboardInterrupt:
        pass
    finally:
        rde_subscriber.destroy_node()
        rclpy.shutdown()


if __name__ == '__main__':
    main()
